docs/CONTRIBUTING.md

```diff
@@ -1,18 +1,238 @@
+   features took years of discussion, rework, testing, and
+   documentation prior to being merged into the master branch.
+
+   All new modules, config options, commands, command parameters, and
+   documents should have "high impact". We do not want to burden users
+   with options that they can not reasonably configure nor do we want
+   to burden them with options that don't provide a notable benefit.
+
+   A reviewer may ask for clarification on how a user is to configure
+   an option - an ideal response will contain details on the process -
+   for example, "users of the MegaX500 are expected to set option X to
+   99.3 while users of the Elite100Y are expected to calibrate option
+   X using procedure ...".
+
+   If the goal of an option is to make the code more modular then
+   prefer using code constants instead of user facing config options.
+
+   New modules, new options, and new parameters should not provide
+   similar functionality to existing modules - if the differences are
+   arbitrary than it's preferable to utilize the existing system or
+   refactor the existing code.
+
+Klipper does not implement a strict "coding style guide", but
+modifications to existing code should follow the high-level code flow,
+code indentation style, and format of that existing code. Submissions
+of new modules and systems have more flexibility in coding style, but
+it is preferable for that new code to follow an internally consistent
+style and to generally follow industry wide coding norms.
+
+It is not a goal of a review to discuss "better implementations".
+However, if a reviewer struggles to understand the implementation of a
+submission, then they may ask for changes to make the implementation
+more transparent. In particular, if reviewers can not convince
+themselves that a submission is free of defects then changes may be
+necessary.
+
+As part of a review, a reviewer may create an alternate Pull Request
+for a topic. This may be done to avoid excessive "back and forth" on
+minor procedural items and thus stream-line the submission process. It
+may also be done because the discussion inspires a reviewer to build
+an alternative implementation. Both situations are a normal result of
+a review and should not be considered criticism of the original
+submission.
+
+### Reviewers
+
+The Klipper "reviewers" are:
+
+| Name                   | GitHub name       | Areas of interest |
+| ---------------------- | ----------------- | ----------------- |
+| Eric Callahan          | @Arksine          | Bed leveling, MCU flashing |
+| Dmitry Butyugin        | @dmbutyugin       | Input shaping, resonance testing, kinematics |
+| Kevin O'Connor         | @KevinOConnor     | Core motion system, Micro-controller code |
+
+Please do not "ping" any of the reviewers and please do not direct
+submissions at them. All of the reviewers monitor the forums and PRs,
+and will take on reviews when they have time to.
```

Comment:
If we have been working on a PR, and once its deemed complete, how do we notify anyone that it's worth their time to actually pay attention?

Adding a tag/label would be about the right level of visibility, but Github doesn't have enough permission granularity that would allow the random person to set a few tags, but not others.

Starting all pr's in draft state could help a bit, but I think it means different enough things to different people that it isn't a solution in itself.

Reply:
The answer is to use GitHub Actions to auto-tag a PR once it is done.

There are many ways to skin this cat; but, having a way that an author can make their changes as "Now ready for Review", usually in a comment with a special keyword, works best.

The overall process works best by auto-warning that their PR is about to be closed, and then auto-closing after a set time (30 days, 90 days, etc).

That way, the author is tagged to update the PR on the next few months.

Taking the burden off of the reviewers